### DISCUSSION TITLE
Terdapat `Warning: glm.fit: algorithm did not converge`

### DISCUSSION THREAD

**Discussion by yuliristiana** [teamalgoritma/community#766] [Q&A] [answered]
Saat ini saya sedang mengalami kendala warning ketika mengulang dan menjalankan program dari yang sebelumnya telah saya buat.  

Muncul warning nya seperti dibawah ini :
![Screenshot 2023-09-10 224747](https://github.com/teamalgoritma/community/assets/134671802/4a3a8862-ff67-4cb7-b234-6155d171dfbc)

Dan juga ketika saya check hasil outputnya ada columns yang NA sebanyak 3. Padahal ketika proses data wrangling untuk saya sudah coba hilangkan dengan menggunakan fungsi `replace`.
> PCOS <- replace(PCOS, is.na(PCOS), 0) <

Apakah hal tersebut yang menjadikan data output saya mengalami muncul warning?
Dan apakah untuk tersebut akan berpengaruh pada proses selanjutanya atau hasilnya nanti?
Perlukah dihilangkan? menggunakan fungsi apa?

terima kasih!

**Comment by kinantytasyao** [comment, marked as answer]
Warning tersebut menandakan bahwa terjadi Perfect Separator, artinya terdapat prediktor yang dapat membedakan target variabel PCOS yes atau no dengan sangat jelas tanpa perlu machine learning. Apakah output columns yang NA maksudnya adalah koefisien regresinya Bu? Jika iya maka kemungkinan besar kolom2 memiliki multikolinearitas

Demikian yang dapat kami sampaikan. Semoga membantu!

**Comment by yuliristiana** [comment]
Hai mba Kinan,
lebih tepatnya muncul otputnya seperti ini sih mba
![image](https://github.com/teamalgoritma/community/assets/134671802/12a3970d-9894-4626-ba0e-b4a4ac6cd8ef)

apakah hal tersebut bisa dikatakan koefisien regresi mba?
dan selanjutnya, sebaiknya hal yang harus saya lakukan mengabaikannya saja atau bagaimana mba?

**Reply by yuliristiana** [reply]
Dan ketika saya coba untuk menjalankan dan menggunakan fungsi step, muncul outputnya seperti ini mba Kinan

![image](https://github.com/teamalgoritma/community/assets/134671802/bbebb878-f90f-4c88-a77d-5a5618431b8a)

**Reply by kinantytasyao** [reply]
Betul Bu yg NA adalah koefisien regresinya dan berarti terdapat multikolinearitas atau prediktor Ibu saling memiliki korelasi, sepertinya korelasi antar variabel Beta.HCG itu besar. dan lagi2 warning glm.fit: algorithm did not converge di fungsi step() menandakan terdapat variabel prediktor yg berperan sebagai perfect separator. Sebaiknya Ibu melakukan feature selection dulu untuk menghilangkan multikolinearitas dan mengeliminasi variabel prediktor yang berperan sebagai perfect separator.

**Reply by yuliristiana** [reply]
Boleh minta tolong gak mba share step cara melakukan feature selection. Karena saya coba pelajari dan terapkan dari beberapa sumber masih juga error ketika saya coba aplikasikan ke data yang saya punya.

**Reply by kinantytasyao** [reply]
Setelah dicocokan dengan data yang Ibu berikan, sepertinya permasalahannya terletak pada belum digantinya data type pada variabel-variabel yang ada. Silakan pastikan terlebih dahulu bahwa data dtype sudah benar sebelum memasukkannya ke dalam model machine learning ya Bu :)

**Reply by yuliristiana** [reply]
Hai Mba Kinan, untuk sarannya saya sudah coba follow.  Dengan hasil tipe data sebagai berikut :
![image](https://github.com/teamalgoritma/community/assets/134671802/fbe89400-4ae4-4e54-ad59-082bd3e8623a)

Namun ketika saya coba untuk menjalankan lagi di bagian modeling, terjadi muncul error kembali.
![image](https://github.com/teamalgoritma/community/assets/134671802/aadb7d6a-258e-424c-b5aa-d576b0b78f26)

Apakah masih perlu dilakukan feature selection dahulu? Jika iya, boleh minta tolong bantu share step nya mba..

**Reply by yuliristiana** [reply]
Dari beberapa channel link yang saya coba dan untuk step feature selection terakhir yang saya coba follow dari link ini mba Kinan. Tapi masih belum ada yang membuahkan hasil baik.

https://rpubs.com/ohcsnad/feature_selection_methods#:~:text=Feature%20selection%20method%20is%20an,without%20losing%20too%20much%20information.

Di saya ketika coba folow munculnya seperti ini, hanya muncul 1 baris saja. 
Note :  **tidak ada columns FSH/LH.**
![image](https://github.com/teamalgoritma/community/assets/134671802/63eb1251-82cf-4c7b-9f10-9c2750085f42)

Dan yang ini ada columns FSH/LH.
![image](https://github.com/teamalgoritma/community/assets/134671802/f850c0d8-a30c-464e-9808-a02452b9f1ee)

**Comment by ranydwi** [comment]
Dear Bu @yuliristiana ,

Terima kasih untuk kesediannya menunggu. Seperti yang sudah disarankan oleh Mba @kinantytasyao , error yang muncul seperti ini:

![image](https://github.com/teamalgoritma/community/assets/68055107/949b6f79-0855-46ce-8ef0-0aa41a553df8)

disebabkan karena tipe data kelas target Ibu belum dirubah menjadi `factor` silahkan ganti terlebih dahulu untuk tipe data kelas targetnya, baru jalankan kembali syntax yang dimaksud.

Kemudian, untuk error `object 'FSH' not found` disebabkan karena sepertinya data Bu Yuli tidak mengandung variabel itu, coba diteliti kembali Bu. Penulisannya saya rasa perlu direname saja untuk variabel yang bersangkutan menjadi `FSH_ratio_LH` agar bisa dibaca oleh R dengan baik.

Thank you

Regards,
Rany

**Comment by ramdhinafinita** [comment]
Halo Ibu @yuliristiana, untuk menjaga agar forum diskusi berjalan kondusif, kami izin melakukan close pada diskusi ini karena belum ada respon setelah 7 hari respon terakhir dari kami. Semoga dapat dipahami, terimakasih.
